pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -100,6 +109,73 @@ func validateResourceGroup(client *Client, ic *types.InstallConfig, path *field.
 	return append(allErrs, field.NotFound(path.Child("resourceGroupID"), ic.AlibabaCloud.ResourceGroupID))
 }
 
+func validateVpc(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vpcs, err := client.ListVpcs(ic.AlibabaCloud.VpcID)
+	if err != nil {
+		return append(allErrs, field.InternalError(path.Child("vpcID"), err))
+	}
+	if vpcs.TotalCount == 0 {
+		allErrs = append(allErrs, field.NotFound(path.Child("vpcID"), ic.AlibabaCloud.VpcID))
+	}
+	return allErrs
+}
+
+func validateVSwitches(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	zoneIDs := map[string]bool{}
+	for idx, id := range ic.AlibabaCloud.VSwitchIDs {
+		vswitches, err := client.ListVSwitches(id)
+		if err != nil {
+			return append(allErrs, field.InternalError(path.Child("vswitchIDs"), err))
+		}
+		if vswitches.TotalCount != 1 {
+			allErrs = append(allErrs, field.NotFound(path.Child("vswitchIDs").Index(idx), id))
+			continue
+		}
+
+		vswitch := vswitches.VSwitches.VSwitch[0]
+		if vswitch.VpcId != ic.AlibabaCloud.VpcID {
+			allErrs = append(allErrs, field.Invalid(path.Child("vswitchIDs").Index(idx), id, fmt.Sprintf("the VSwitch does not belong to vpc %s", ic.AlibabaCloud.VpcID)))
+		}
+
+		if zoneIDs[vswitch.ZoneId] {
+			allErrs = append(allErrs, field.Invalid(path.Child("vswitchIDs").Index(idx), id, "the availability zone of the VSwitch overlapped with other VSwitches"))
+		} else {
+			zoneIDs[vswitch.ZoneId] = true
+		}
+	}
+
+	return allErrs
+}
+
+func validatePrivateZoneID(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	fldpath := path.Child("privateZoneID")
+	zoneName := ic.ClusterDomain()
+	isExist := false
+
+	zones, err := client.ListPrivateZones("", ic.AlibabaCloud.VpcID)
```

Comment:
Doh. I missed that `QueryVpcId` was VPC ID instead of Zone ID. Nevertheless, I think this is fine.